src/Form.js

```diff
@@ -29,6 +41,7 @@ const Form = () => {
   const DEFAULT_ERROR_TEXT = 'Something went wrong, please try again later';
 
   const [loading, setLoading] = useState(false);
+  const [hasDownloadStarted, setHasDownloadStarted] = useState(false);
```

Comment:
Nit: No need to change this now, but here's a super interesting read on why not to use boolean flags for tracking state and how to do it instead to avoid conflicting flags: https://dev.to/davidkpiano/no-disabling-a-button-is-not-app-logic-598i